templates/digital-asset-template/frontend/hooks/useGetCollections.tsx

```diff
@@ -0,0 +1,93 @@
+import { aptosClient } from "@/utils/aptosClient";
+import { AccountAddress } from "@aptos-labs/ts-sdk";
+import { useState, useEffect } from "react";
+
+/**
+ * A react hook to get all collections under the current contract.
+ *
+ * This call can be pretty expensive when fetching a big number of collections,
+ * therefore it is not recommended to use it in production
+ *
+ */
+export function useGetCollections() {
+  const [fas, setFAs] = useState<any[]>([]);
+
+  useEffect(() => {
+    // fetch the contract registry address
+    getRegistry().then((registry) => {
+      // fetch collections objects created under that contract registry address
+      getObjects(registry).then((objects) => {
+        // get each collection object data
+        getCollections(objects).then((data) => {
+          setFAs(data);
+        });
+      });
+    });
+  }, []);
+
+  return fas;
+}
```

Comment:
Might be worth it to use `react-query` here after my PR lands. It will handle caching this for you, and you can specify not to fetch unless you invalidate it.